// @octokit/oauth-methods/dist-src/scope-token.js
import { request as defaultRequest } from "@octokit/request";
import btoa from "btoa-lite";
export async function scopeToken(options) {
    const { request: optionsRequest, clientType, clientId, clientSecret, token, ...requestOptions } = options;
    const request = optionsRequest ||
        /* istanbul ignore next: we always pass a custom request in tests */
        defaultRequest;
    const response = await request("POST /applications/{client_id}/token/scoped", 
    // @ts-expect-error - TODO: I don't get why TS is complaining here. It works with `defaultRequest` directly
    {
        headers: {
            authorization: `basic ${btoa(`${clientId}:${clientSecret}`)}`,
        },
        client_id: clientId,
        access_token: token,
        ...requestOptions,
    });
    const authentication = Object.assign({
        clientType,
        clientId,
        clientSecret,
        token: response.data.token,
    }, response.data.expires_at ? { expiresAt: response.data.expires_at } : {});
    // @ts-expect-error - response.status type is incompatible (200 vs number)
    return { ...response, authentication };
}
